File: backend/internal/model/task.go
package model

import "time"

const TaskStatusCompleted = "完了"

type Task struct {
	ID               uint       `gorm:"primaryKey" json:"id"`
	BacklogIssueID   int        `gorm:"index;default:0" json:"backlogIssueId"`
	ParentIssueID    int        `gorm:"index;default:0" json:"parentIssueId"`
	IssueKey         string     `gorm:"uniqueIndex;not null" json:"issueKey"`
	Title            string     `gorm:"not null" json:"title"`
	Description      string     `json:"description"`
	Priority         string     `json:"priority"`
	EstimatedHours   float64    `gorm:"default:0" json:"estimatedHours"`
	DueDate          *time.Time `json:"dueDate"`
	Status           string     `json:"status"`
	AssigneeID       int        `gorm:"index;default:0" json:"assigneeId"`
	SpaceID          uint       `gorm:"index;not null" json:"spaceId"`
	Score            float64    `gorm:"default:0" json:"score"`
	MilestoneID      string     `json:"milestoneId"`
	MilestoneDueDate *time.Time `json:"milestoneDueDate"`
	BacklogCreatedAt    *time.Time `json:"backlogCreatedAt"`
	CreatedUserName     string     `json:"createdUserName"`
	CreatedUserIconURL  string     `json:"createdUserIconUrl"`
	SyncedAt         time.Time  `json:"syncedAt"`
	LastNotifiedAt   *time.Time `json:"lastNotifiedAt"`
	IsWatched        bool       `gorm:"index;default:false" json:"isWatched"`
	CreatedAt        time.Time  `json:"createdAt"`
	UpdatedAt        time.Time  `json:"updatedAt"`
	Categories       []Category `gorm:"many2many:task_categories" json:"categories,omitempty"`
}

File: src/renderer/src/types/Task.ts
export interface Task {
  id: number
  issueKey: string
  title: string
  description: string
  priority: string
  estimatedHours: number
  dueDate: string | null
  status: string
  score: number
  spaceId: number
  milestoneId: string
  milestoneDueDate: string | null
  backlogCreatedAt: string | null
  createdUserName: string
  createdUserIconUrl: string
  isWatched: boolean
}

export interface Space {
  id: number
  domain: string
  displayName: string
  color: string
  isActive: boolean
}

export interface Memo {
  id: number
  taskId: number
  content: string
  createdAt: string
  updatedAt: string
}

export interface PersonalTask {
  id: number
  title: string
  description: string
  startDate: string | null
  dueDate: string | null
  estimatedHours: number
  isCompleted: boolean
  completedAt: string | null
  parentBacklogTaskId: number | null
  orderIndex: number
  createdAt: string
  updatedAt: string
}

export interface PersonalTaskInput {
  title: string
  description?: string
  startDate?: string | null
  dueDate?: string | null
  estimatedHours?: number
  parentBacklogTaskId?: number | null
  isCompleted?: boolean
}
